### PR TITLE
asm using keystone

### PR DESCRIPTION
**Pull request by david942j**
[diff with #20 ](https://github.com/peter50216/pwntools-ruby/compare/feature/disasm...feature/asm)
supported i386 and amd64

---
This change is [Reviewable](https://reviewable.io/reviews/peter50216/pwntools-ruby/24)

### CODE REVIEW

File: travis/install.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+setup_osx()
+{
+  # install capstone
+  brew install capstone
+  export DYLD_LIBRARY_PATH=/usr/local/opt/capstone/lib:$DYLD_LIBRARY_PATH
+
+  # install keystone
+  install_keystone_from_source
+  cp keystone/build/llvm/lib/libkeystone.dylib . # hack, don't know why
```
Comment (david942j) [inline travis/install.sh]: don't know why the next line `export DYLD_LIBRARY_PATH=$TRAVIS_BUILD_DIR/keystone/build/llvm/lib:$DYLD_LIBRARY_PATH` has no effect

**Comment by hanhanW** [comment]
There are some conflicts, please fix it.

**Comment by hanhanW** [comment]
There are some conflicts again QQ

**Comment by hanhanW** [comment]
Reviewed 2 of 64 files at r3, 5 of 57 files at r4, 2 of 9 files at r5, 3 of 4 files at r6.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[Gemfile, line 7 at r6](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/24#-KsfmDcAG1WRzGQ6wjAd:-KsfmDcAG1WRzGQ6wjAe:b-w18izh) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/dccbd52cbae3cc63276ed9e8ec9beba78e18c04d/Gemfile#L7)):*
> ```Ruby
> gem 'crabstone', git: 'https://github.com/bnagy/crabstone' # it doesn't push to RubyGems :(
> 
> gem 'keystone', git: 'https://github.com/sashs/ruby-keystone'
> ```

I think it didn't push to RubyGems either. Please request for pushing the gem.

---

*[lib/pwnlib/asm.rb, line 74 at r6](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/24#-KshDSjNOrAzWFKqi_ht:-KshDSjOy17nXUDzQpLS:b-jg6itq) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/dccbd52cbae3cc63276ed9e8ec9beba78e18c04d/lib/pwnlib/asm.rb#L74)):*
> asm(shellcraft.amd64.linux.sh)

context.local(arch: 'amd64') { asm(shellcraft.amd64.linux.sh) }

---

*[test/asm_test.rb, line 17 at r6](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/24#-KshGu7fJcedl-VUj2rw:-KshGu7fJcedl-VUj2rx:b-jjzptt) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/dccbd52cbae3cc63276ed9e8ec9beba78e18c04d/test/asm_test.rb#L17)):*
> ```Ruby
>   end
> 
>   def test_i386_asm
> ```

use ()

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/24)*

**Comment by hanhanW** [comment]
:lgtm:
LGTM

---

Reviewed 3 of 3 files at r7.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/24#-:-KtBcuTelduEDiB1Ml2G:bauglql)*

**Pull request merged**